internal/storage/image.go

```diff
@@ -420,6 +438,209 @@ func (svc *imageService) imageStatus(systemContext *types.SystemContext, unstabl
 	return &result, nil
 }
 
+func (svc *imageService) PullLocalManifestDigest(ctx context.Context, systemContext *types.SystemContext, userSpecifiedImage string) (string, error) {
+	imageParsed, err := reference.ParseNormalizedNamed(userSpecifiedImage)
+	if err != nil {
+		return "", fmt.Errorf("parsing user specified image: %w", err)
+	}
+	imageParsed = reference.TagNameOnly(imageParsed)
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return "", fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return "", fmt.Errorf("create policy context: %w", err)
+	}
+
+	defer func() {
+		if err := policyContext.Destroy(); err != nil {
+			log.Errorf(ctx, "Unable to destroy policy: %v", err)
+		}
+	}()
+
+	localManifestDigest, err := svc.pullLocalManifestDigest(ctx, systemContext, policyContext, imageParsed)
+	if err != nil {
+		return "", fmt.Errorf("pull local manifest digest: %w", err)
+	}
+	log.Debugf(ctx, "Local manifest digest for image %q is: %s", imageParsed, localManifestDigest)
+
+	canonicalRef, err := reference.WithDigest(reference.TrimNamed(imageParsed), localManifestDigest)
+	if err != nil {
+		return "", fmt.Errorf("create canonical reference: %w", err)
+	}
+
+	return canonicalRef.String(), nil
+}
+
+func (svc *imageService) pullLocalManifestDigest(ctx context.Context, sys *types.SystemContext, policyContext *signature.PolicyContext, imageParsed reference.Named) (digest.Digest, error) {
+	srcRef, err := docker.NewReference(imageParsed)
+	if err != nil {
+		return "", fmt.Errorf("creating docker:// reference for %q: %w", imageParsed.String(), err)
+	}
+
+	destRef, err := istorage.Transport.NewStoreReference(svc.store, imageParsed, "")
+	if err != nil {
+		return "", fmt.Errorf("creating containers-storage: reference for %q: %w", imageParsed.String(), err)
+	}
+
+	manifestBytes, err := copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
```

Comment:
On the `IsRunningImageAllowed` caller at least, no credentials available = can’t be relied upon.